docs/standard/serialization/system-text-json-migrate-from-newtonsoft-how-to.md

```diff
@@ -0,0 +1,453 @@
+* Returns the stored complex values as a `JObject` or `JArray`. *Complex values* are collections of JSON key-value pairs within braces (`{}` ) or lists of values within brackets (`[]`). The properties and values within the braces or brackets can have additional properties or values.
+* Returns a `null` reference when the payload has the `null` JSON literal.
+
+`System.Text.Json` stores a boxed `JsonElement` for both primitive and complex values within the `System.Object` property or dictionary value. However, it treats `null` the same as `Newtonsoft.Json` and returns a null reference when the payload has the `null` JSON literal in it.
+
+To implement type inference for `Object` properties, create a converter like the example in [How to write custom converters](system-text-json-converters-how-to.md#deserialize-inferred-types-to-object-properties).
+
+### Maximum depth
+
+`Newtonsoft.Json` doesn't have a maximum depth limit. For `System.Text.Json` there's a default limit  of 64, and it's configurable by setting (xref:System.Text.Json.JsonSerializerOptions.MaxDepth?displayProperty=nameWithType).
+
+## Scenarios using JsonSerializer that require workarounds
+
+The following scenarios aren't supported by built-in functionality, but sample code is provided for workarounds. Most of the workarounds require that you implement [custom converters](system-text-json-converters-how-to.md).
+
+### Specify date format
+
+`Newtonsoft.Json` provides a `DateTimeZoneHandling` option that lets you specify the serialization format used for `DateTime` and `DateTimeOffset` properties. In `System.Text.Json`, the only format that has built-in support is ISO 8601-1:2019 since it is widely adopted, unambiguous, and makes round trips precisely. To use any other format, create a custom converter. For more information, see [DateTime and DateTimeOffset support in System.Text.Json](../datetime/system-text-json-support.md).
+
+### Quoted numbers
+
+`Newtonsoft.Json` can serialize or deserialize numbers represented by JSON strings (surrounded by quotes). For example, it can accept: `{ "DegreesCelsius":"23" }` instead of `{ "DegreesCelsius":23 }`. To enable that behavior in `System.Text.Json`, implement a custom converter like the following example. The converter handles properties defined as `long`:
+
+* It serializes them as JSON strings. 
+* It accepts JSON numbers and numbers within quotes while deserializing.
+
+[!code-csharp[](~/samples/snippets/core/system-text-json/csharp/LongToStringConverter.cs)]
+
+Register this custom converter by [using an attribute](system-text-json-converters-how-to.md#registration-sample---jsonconverter-on-a-property) on individual `long` properties or by [adding the converter to the `Converters` collection](system-text-json-converters-how-to.md#registration-sample---converters-collection).
```

Comment:
Should we link to the `Converters` API instead here rather than the doc?

I would think some of these samples would actually end up in the API docs too, no? Maybe we update the link once this doc also has a copy of the sample usage?
Here: https://docs.microsoft.com/en-us/dotnet/api/system.text.json.jsonserializeroptions.converters?view=netcore-3.1

Reply:
> Should we link to the `Converters` API instead here rather than the doc?

On second thought, looks like we have this text all over the place. Pointing to the doc is fine.

What about including both links?

```suggestion
Register this custom converter by [using an attribute](system-text-json-converters-how-to.md#registration-sample---jsonconverter-on-a-property) on individual `long` properties or by [adding the converter](system-text-json-converters-how-to.md#registration-sample---converters-collection) to the [`Converters`](https://docs.microsoft.com/en-us/dotnet/api/system.text.json.jsonserializeroptions.converters?view=netcore-3.1) collection.
```